src/datafile.rs

```diff
@@ -180,13 +178,8 @@ impl Iterator for DataFileIterator {
 
     fn next(&mut self) -> Option<Self::Item> {
         let offset = self.file.seek(SeekFrom::Current(0)).unwrap();
-
         let decoded_maybe = bincode::deserialize_from(&self.file);
-        if let Err(_) = decoded_maybe {
-            return None;
-        }
-
-        Some((offset, decoded_maybe.unwrap()))
+        Some((offset, decoded_maybe.ok()?))
```

Comment:
```rust
let foo : Result<_, _> = ..;
if let Err(..) = foo {
    return None;
}
Some(foo.unwrap())
```

this can simply be done as using `Result::ok()` to return it into an `Option` and then using `?` to safely 'unwrap' it.

```rust
let foo : Result<_, _> = ..;
Some((other, foo.ok()?))
```